File: src/Human.java
import java.util.Random;

public class Human {
    private String eMD;
    private int age;
    private double currentSpeed;
    private double x, y; // координаты


    public Human(String eMD, int age, double currentSpeed) {
        this.eMD = eMD;
        this.age = age;
        this.currentSpeed = currentSpeed;
        this.x = 0.0;
        this.y = 0.0;
    }


    public String getEMD() { return eMD; }
    public void setEMD(String eMD) { this.eMD = eMD; }

    public int getAge() { return age; }
    public void setAge(int age) { this.age = age; }

    public double getCurrentSpeed() { return currentSpeed; }
    public void setCurrentSpeed(double currentSpeed) { this.currentSpeed = currentSpeed; }

    public double getX() { return x; }
    public double getY() { return y; }


    public void move() {
        Random rand = new Random();
        double angle = rand.nextDouble() * 2 * Math.PI;
        double distance = this.currentSpeed;

        this.x += distance * Math.cos(angle);
        this.y += distance * Math.sin(angle);
    }
}

File: src/Human.kt
import kotlin.math.cos
import kotlin.math.sin
import kotlin.math.PI

class Human(
    override val fullName: String,
    private var age: Int,
    override var currentSpeed: Double
) : Movable {
    override var x: Double = 0.0
    override var y: Double = 0.0

    fun getAge(): Int = age
    fun setAge(newAge: Int) { age = newAge }
    fun setCurrentSpeed(newSpeed: Double) { currentSpeed = newSpeed }

    override fun move() {
        val direction = kotlin.random.Random.nextDouble() * 2 * PI
        val dx = currentSpeed * cos(direction)
        val dy = currentSpeed * sin(direction)
        x += dx
        y += dy
        println("$fullName переместился в (${"%.2f".format(x)}, ${"%.2f".format(y)})")
    }

    override fun getPosition(): Pair<Double, Double> = Pair(x, y)
}